cuda/base/math.hpp

```diff
@@ -93,17 +94,126 @@ __device__ GKO_INLINE std::complex<double> one<std::complex<double>>()
 }
 
 
-// Since the `isfinite` function from CUDA is not in any namespace, it
-// will be used without the `using` keyword.
+// This first part is specific for clang in combination with the nvcc compiler
```

Comment:
In the end, this is also required for Intel? Maybe you could also mention it, then.

Reply:
Actually, I do not remember why I also added Intel in the defines, I will have to check if it also happens there. If it does, I will add it in the comments.

Reply:
I tested it with the intel compiler, and apparently, it does not have the same issue as the clang compiler has.
Thanks for finding this discrepancy! I removed the intel `defined` statements.

Reply:
OK, I guess I messed up the testing, since it fails for intel now on the CI. I guess I should have set the `CUDA_HOST_COMPILER`.
I will add the intel compiler to the comments here.